### DISCUSSION TITLE
Which pooling method should I choose?

### DISCUSSION THREAD

**Discussion by ZhangGongjie** [pyg-team/pytorch_geometric#2868] [General]
Hi, I am a newbie in GNNs, and I wish to apply GNNs as a tool to address my specific problems. I know there are many pooling methods, but I don't know which one to use.

Here is the setting: I build a weighted undirected graph, which has nodes of (num_node=1024, feat_dim=256) with approximately 10k edges. Of these nodes, many are redundant (can be expressed with adjacent nodes). If I wish to remove these redundant nodes that can be expressed with adjacent nodes, which graph pooling method should I choose? Memory and computational cost is also an important factor here.

Thanks, everyone.

**Comment by rusty1s** [comment]
I think you wanna use the `TopKPooling` approach, which tries to filter out redundant nodes by design.

**Reply by ZhangGongjie** [reply]
Thank you!  SAGPool, ASAPool and TopKPool all belong to this category, right? Any recommendations?

**Reply by rusty1s** [reply]
That is correct. All these approaches are quite similar, so I would probably start with a `TopKPooling` approach to see if it already fulfills your needs.

**Reply by ZhangGongjie** [reply]
Thanks!
